src/Symfony/Bundle/TwigBundle/Cache/ChainCache.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\Cache;
+
+use Twig\Cache\CacheInterface;
+
+/**
+ * Chains several caches together.
+ *
+ * Cached items are fetched from the first cache having them in its data store.
+ * They are saved and deleted in all adapters at once.
+ */
+class ChainCache implements CacheInterface
+{
+    /**
+     * @param CacheInterface[] $caches The ordered list of caches used to store and fetch cached items
+     */
+    public function __construct(
+        private array $caches,
+    ) {
+        if (!$caches) {
+            throw new \InvalidArgumentException('At least one cache must be specified.');
+        }
+
+        foreach ($caches as $cache) {
+            if (!$cache instanceof CacheInterface) {
+                throw new \InvalidArgumentException(sprintf('The class "%s" does not implement the "%s" interface.', get_debug_type($cache), CacheInterface::class));
+            }
+        }
+    }
+
+    public function generateKey(string $name, string $className): string
+    {
+        return $className.'#'.$name;
+    }
+
+    /**
+     * @return array{name: string, className: string}
+     */
+    protected function splitKey(string $key): array
```

Comment:
This method has logic that should stay internal to allow any change in case we find an optimization or a but to fix. I would make it `private`.

```suggestion
    private function splitKey(string $key): array
```

An maybe make this class `final`. It's still possible to create a decorator if someone want to inherit par of its behavior.

Reply:
this should indeed be a private method